### PR TITLE
Migrate lift table to datagrid

### PR DESCRIPTION
**Pull request by Angatupyry**
## What's new

- Migrate lift table to datagrid mui
- New test file was added
- New storybook file was added
- Removed the use of the basic table and instead used mui's datagrid.

[chrome-capture-2023-5-5.webm](https://github.com/open-rmf/rmf-web/assets/37310205/f33c30a8-ed52-4150-b940-b488b9276870)

## Self-checks

- [ ] I have prototyped this new feature (if necessary) on Figma 
- [ ] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components
- [ ] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test

### CODE REVIEW

File: packages/react-components/lib/lifts/lift-table-datagrid.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  return (
+    <div style={{ height: '100%', width: '100%' }}>
+      <DataGrid
+        autoHeight={true}
+        getRowId={(l) => l.index}
+        rows={lifts}
+        pageSize={5}
+        rowHeight={38}
+        columns={columns}
+        rowsPerPageOptions={[5]}
```
Comment (aaronchongth) [inline packages/react-components/lib/lifts/lift-table-datagrid.tsx:157]:
```suggestion
        rowsPerPageOptions={[5]}
        localeText={{
          noRowsLabel: 'No lifts available.',
        }}
```

File: packages/react-components/lib/lifts/lift-table-datagrid.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      field: 'name',
+      headerName: 'Name',
+      width: 90,
+      valueGetter: (params: GridValueGetterParams) => params.row.name,
+      flex: 1,
+      filterable: true,
+    },
+    {
+      field: 'currentFloor',
+      headerName: 'Current floor',
```
Comment (aaronchongth) [inline packages/react-components/lib/lifts/lift-table-datagrid.tsx:110]:
```suggestion
      headerName: 'Current Floor',
```
nit: similar to the review comment on doors, we should capitalize the second word for the column headings

**Review by aaronchongth — CHANGES_REQUESTED** [review]
Thanks! Just a few comments similar to the ones for doors.
